### ISSUE TITLE
CommonJS detection fails

### ISSUE THREAD

**Issue by tindzk** [scalacenter/bloop#715]
While working on #634, I noticed that the sbt plug-in fails to detect CommonJS modules. I could trace it down to this function in `SbtBloop.scala`:

```scala
  lazy val findOutScalaJsStage: Def.Initialize[Option[String]] = Def.settingDyn {
    try {
      val stageClass = Class.forName("org.scalajs.sbtplugin.Stage")
      val stageSetting = proxyForSetting("scalaJSStage", stageClass)
      Def.setting {
        stageSetting.value.toString match {
          case "Some(FastOpt)" => Some(ScalaJsFastOpt)
          case "Some(FullOpt)" => Some(ScalaJsFullOpt)
          case _ => None
        }
      }
    } catch {
      case _: ClassNotFoundException => Def.setting(None)
    }
  }
```

A `ClassNotFoundException` gets triggered. When I changed `stageClass` to `classOf[AnyRef]`, `stageSetting.value` would always be `None`.

**Comment by jvican** [comment]
Good catch. I don't really understand what you mean by changed `stageClass` to `classOf[AnyRef]`. Also, I struggle to see how this is related to common js, would you mind elaborating on that point? :smile:
  
On commonjs, will we need to work on support for `commonJSName` too? https://www.scala-js.org/doc/project/dependencies.html

**Comment by jvican** [comment]
> commonJSName is only used for jsDependencies which we do not need to support in Bloop.
  
How is the linker able to detect that a certain jar is mapped to a `commonJSName` to create the prelude explained in the docs? https://www.scala-js.org/doc/project/dependencies.html

**Comment by tindzk** [comment]
`commonJSName` was originally part of the Scala.js sbt plug-in, but was moved into a separate project called [jsdependencies](https://github.com/scala-js/jsdependencies). You can find the implementation [here](https://github.com/scala-js/jsdependencies/blob/804501756ec689e90075ed3836b4d9cc31bd66aa/jsdependencies-sbt-plugin/src/main/scala/org/scalajs/jsdependencies/sbtplugin/JSDependenciesPlugin.scala#L386-L408).

Please note that the README states the following:

> sbt-jsdependencies is not particularly recommended for new projects. Most projects should use scalajs-bundler instead, which provides saner dependencies through npm.

**Comment by jvican** [comment]
I think this is partially solved by https://github.com/scalacenter/bloop/pull/714/files which hasn't yet been picked up for me since I abandoned the PR a few weeks ago. I will probably continue the work I started in a matter of weeks as other work items have more priority.
